atd-vze/src/views/Locations/LocationNotes.js

```diff
@@ -0,0 +1,257 @@
+import React, { useState } from "react";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import { Card, CardHeader, CardBody, CardFooter, Table, Input, Button } from "reactstrap";
+import moment from "moment";
+import { locationNotesDataMap } from "./locationNotesDataMap";
+import { GET_LOCATION_NOTES, INSERT_LOCATION_NOTE, UPDATE_LOCATION_NOTE, DELETE_LOCATION_NOTE } from "../../queries/locationNotes";
+import { useAuth0, isReadOnly } from "../../auth/authContext";
+
+// declare a notes component
+const LocationNotes = ({ locationId }) => {
+
+  // add a state variable to manage value when new note is entered
+  const [newNote, setNewNote] = useState("");
+  const [editedNote, setEditedNote] = useState("");
+  const [editRow, setEditRow] = useState("");
+
+  // disable edit features if role is "readonly"
+  const { getRoles } = useAuth0();
+  const roles = getRoles();
+
+  // get current users email
+  const userEmail = localStorage.getItem("hasura_user_email");
+
+  // fetch data from database using graphQL query
+  const { loading, error, data, refetch } = useQuery(GET_LOCATION_NOTES, {
+    variables: { locationId },
+  });
+
+  // declare mutation functions
+  const [addNote] = useMutation(INSERT_LOCATION_NOTE);
+  const [editNote] = useMutation(UPDATE_LOCATION_NOTE);
+  const [deleteNote] = useMutation(DELETE_LOCATION_NOTE);
+
+  if (loading) return "Loading...";
+  if (error) return `Error! ${error.message}`;
+
+  const tableName = "location_notes";
+  const keyField = "id";
+  const fieldConfig = locationNotesDataMap[0];
+
+  // function to handle add button click
+  const handleAddNoteClick = () => {
+    addNote({
+      variables: {
+        note: newNote,
+        locationId: locationId,
+        userEmail: userEmail
+      }
+    }).then(response => {
+      setNewNote("");
+      refetch();
+    }).catch(error => console.error(error));
+  };
+
+  // function to handle edit button click
+  const handleEditClick = (row) => {
+    setEditedNote(row.text);
+    setEditRow(row);
+  };
+
+  // function to handle save edit button click
+  const handleSaveClick = (row) => {
+    const id = row.id
+    editNote({
+      variables: {
+        note: editedNote,
+        id: id
+      }
+    }).then(response => {
+			refetch().then(response => {
+			setEditedNote("");
+			setEditRow("");
+			})
+    }).catch(error => console.error(error));
+  };
+
+  // function to handle cancel button click
+  const handleCancelClick = () => {
+    setEditedNote("");
+		setEditRow("");
+  };
+
+  // function to handle delete note button click
+  const handleDeleteClick = (row) => {
+    const id = row.id
```

Comment:
should have a semicolon at the end of this line. Prettier would do this for you.